assets/src/editor/events/dates-and-times/data/with-date-ticket-entities.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * External imports
+ */
+import { withSelect } from '@wordpress/data';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { isModelEntityOfModel } from '@eventespresso/validators';
+
+/**
+ * A Hoc exposing ticket entities related to the provided DateTimeEntity.
+ *
+ * @return {function} A higher order component.
+ */
+export const withDateTicketEntities = createHigherOrderComponent(
```

Comment:
Just pointing out the pattern that I'm following here.  The HOC name describes what it does (exposing ticket entities related to a date).  However I'm not exposing `dateTicketEntities` and `dateTicketEntitiesLoaded` but instead just `ticketEntities` and `ticketEntitiesLoaded`.  The reason for this is because it makes the HOC more reusable on components that don't care where `ticketEntities` come from.

Reply:
> why not just withTicketEntities then?

For the same reason outlined in [this comment here](https://github.com/eventespresso/event-espresso-core/pull/1295/files#r292440116).  `withTicketEntities` would either mean to me that it returns `allTicketEntities` available from the server by default (think of some future list table for tickets) or if it accepts an argument that it's a filtered list of tickets by a custom query.

> or `withTicketEntitiesForDate`?

Sure, that's a variation I could use.  I mostly went with `withDateTicketEntities` because that was a shift from `withDatetimeTickets` (the original name).  I didn't even think of using `withTicketEntitiesForDate` to be honest.  

> This is similar to my first comment above in that, in most places you have dropped the extra qualifier from the prop or function name, ie: Date instead of EventDate, so why use DateTicket here?

Also explained (hopefully sufficiently) in [this comment](https://github.com/eventespresso/event-espresso-core/pull/1295/files#r292440116).

> So for example, we might have a HOC named withTicketEntities which would assume you are providing a DateEntity since that would be the default/primary way of retrieving tickets.

I'm aware this is an example you gave, but I think as an example of a pattern you are suggesting we adopt this might not work well.  I think it'd be a mistake to assume `withTicketEntities` always returns tickets belonging to a date.  That requires some domain knowledge about how models work in Event Espresso and the relations.  I do realize that one can look at the argument list to see what the HOC receives, but I'm mostly concerned about future maintainers when they see something like this:

```jsx
export default compose( [
	withInstanceId,
	withFancyUi,
	withTicketEntities,
] )( SomeComponentListingTickets );
```

> Or we could always be explicit in the naming and always reference the source that you are using to retrieve the related entities, so `withTicketEntitiesForDate` or withTicketEntitiesForIds, etc.

For the former, sure we can do that if you want - I just see it as a variation of what is already present (which was mostly just keeping the same order of model names in the original)... but if you want me to go through and rename things to follow that pattern I can.  Regarding `withTicketEntitiesForIds` though, I think that's too vague.  What ids?  Given ticket ids?  Provided datetime ids?  provided registration ids?